cms/static/js/models/group_configuration.js

```diff
@@ -87,23 +88,35 @@ function(Backbone, _, str, gettext, GroupModel, GroupCollection) {
                 };
             }
 
-            if (attrs.groups.length < 1) {
+            if (!this.canBeEmpty && attrs.groups.length < 1) {
                 return {
                     message: gettext('There must be at least one group.'),
                     attributes: { groups: true }
                 };
             } else {
                 // validate all groups
-                var invalidGroups = [];
+                var validGroups = new Backbone.Collection(),
+                    invalidGroups = new Backbone.Collection();
                 attrs.groups.each(function(group) {
-                    if(!group.isValid()) {
-                        invalidGroups.push(group);
+                    if (!group.isValid()) {
+                        invalidGroups.add(group);
+                    } else {
+                        validGroups.add(group);
                     }
                 });
-                if (!_.isEmpty(invalidGroups)) {
+
+                if (!invalidGroups.isEmpty()) {
                     return {
                         message: gettext('All groups must have a name.'),
-                        attributes: { groups: invalidGroups }
+                        attributes: { groups: invalidGroups.toJSON() }
+                    };
+                }
+
+                var groupNames = validGroups.map(function(group) { return group.get('name'); });
+                if (groupNames.length !== _.uniq(groupNames).length) {
+                    return {
+                        message: gettext('All groups must have a unique name.'),
```

Comment:
@catong - This is the new validation error message for when the user either gives two content groups the same name or two groups within an experiment configuration the same name.